### PR TITLE
feat: Typescript STRICT mode enabled

### PR DESCRIPTION
**Pull request by grod220**
https://www.typescriptlang.org/tsconfig#strict

At the moment, we do not have strict mode enabled in our .tsconfig (https://github.com/sushiswap/sushiswap-interface/blob/master/tsconfig.json#L41). Unfortunately, that means we have a growing number of unsafe code lines (was 1200 violations in October, now it's at 1550) that have crept in silently. In many cases it's issues with nullChecking and implicitAny's. It's also highly likely there are bugs that has slipped past the compiler too.

Enabling strict mode will make us better developers and start making less buggy features. However, it is an impossible task to retroactively fix every one of the issues. This PR adds the in-your-face comment: `@ts-ignore TYPE NEEDS FIXING` to the lines with type violations. Hopefully, as we work on different features that touch files that have these comments, we can fix as we go.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "lib": ["dom", "dom.iterable", "esnext"],
     "allowJs": true,
     "skipLibCheck": true,
-    "strict": false, // in local dev, this should be set to TRUE
+    "strict": true,
```
Comment (grod220) [inline tsconfig.json]: What this PR is all for 🚀

**Comment by matthewlilley** [comment]
Legend

**Comment by matthewlilley** [comment]
Happy to get this in ASAP, but have a few compilation issues by the looks of it at the moment.

**Comment by grod220** [comment]
@matthewlilley fixed! Review? :)

**Review by chillichelli — APPROVED** [review]

**Pull request merged**